### PR TITLE
List logging

### PR DESCRIPTION
**Pull request by shadetree01010100**
This was a good idea and was closed by accident while deleting nio2 branch.

### CODE REVIEW

**Comment by dorzel** [comment]
@mattdodge @tyoungNIO @hansmosh This was originally closed here: https://github.com/nio-blocks/logger/pull/7 What does everyone think of this PR?

**Comment by mattdodge** [comment]
Not sure why it was closed originally, @hansmosh any ideas?

One question I have would be does this actually log the contents of the signal or does it look like this now:
```
[<Signal 0xalkjasef>, <Signal 0xalsdkjre>]
```

**Comment by dorzel** [comment]
@mattdodge According to Tom it was closed accidentally when the nio2 branch was deleted. And woops yes it does indeed log the signal objects instead of the content. Will push up a change

**Comment by hansmosh** [comment]
Do we want to make this a flag so that you have the option to still see signals individually? Pros and Cons for having a flag:

Pros
- New users are not immediately exposed to the fact that signals are actually in lists. Lists could be confusing to a new user.
- The block remains backwards compatible

Cons
- Users are never exposed to the fact that signals are lists until they turn on a flag.
- All the logger block does is log signals. It seems like overcomplicating a simple thing by adding configurable properties to it.

What do you think?
@dorzel @mattdodge @tyoungNIO

**Comment by mattdodge** [comment]
It's a good question. Most of the time when we're logging we just want to see signal structure (i.e., did my data get on there). I would postulate that it's not that common that you want to see "did they all come in the same list or not" but it does happen sometimes. 

I don't have a strong preference either way but I guess if I had to vote I would vote for a hidden/advanced toggle for this and default it to *not* showing lists of signals (i.e., the legacy behavior)

**Comment by dorzel** [comment]
I agree, hidden toggle and default to normal behavior seems like the right solution

**Comment by dorzel** [comment]
@hansmosh @mattdodge @tyoungNIO Changes are pushed up, there is now a hidden property to configure list logging

File: logger_block.py [hunk 1]
```diff
@@ -33,11 +35,16 @@ def process_signals(self, signals):
...
-                self.logger.error("Failed to log signal")
+
+        try:
+            if self.log_as_list():
+                log_func([signal.to_dict() for signal in signals])
+            else:
+                for s in signals:
+                    log_func(s.to_dict())
+        except:
+            self.logger.exception("Failed to log {} signals"
```
Comment (hansmosh) [inline logger_block.py]: This isn't entirely true. If we're logging one signal at a time and it's not the first one that fails, then this number is incorrect.

Comment (dorzel) [reply logger_block.py]: ah forgot to change that

**Comment by dorzel** [comment]
@hansmosh I decided to count the signals successfully logged to get a correct exception message

**Review by hansmosh — APPROVED** [review]

**Pull request merged**